test/acceptance/user-order.controller.acceptance.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createClientForHandler, supertest, expect} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {ShoppingApplication} from '../..';
+import {OrderRepository, UserRepository} from '../../src/repositories';
+import {MongoDataSource} from '../../src/datasources';
+import {User, Order, ShoppingCartItem} from '../../src/models';
+
+describe('UserOrderController acceptance tests', () => {
+  let app: ShoppingApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+  const orderRepo = new OrderRepository(new MongoDataSource());
+  const userRepo = new UserRepository(new MongoDataSource(), orderRepo);
+
+  before(setupApplication);
+  beforeEach(clearDatabase);
+  after(async () => {
+    await app.stop();
+  });
+
+  it('creates an order for a user with a given orderId', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = {
+      orderId: '1',
+      userId: userId,
+      total: 99.99,
+      products: [
+        {
+          productId: 'product1',
+          quantity: 10,
+          price: 9.99,
+        },
+      ],
+    };
+
+    await client
+      .post(`/users/${userId}/orders`)
+      .send(order)
+      .expect(200, order);
+  });
+
+  it('creates an order for a user without a given orderId', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = givenAOrder();
+    order.userId = userId;
```

Comment:
A test-data builder typically accepts a Partial object allowing callers to customize/overwrite default values provided by the builder.

```ts
const order = givenAnOrder({userId});
```